pyfstat/core.py

```diff
@@ -922,48 +911,167 @@ def init_computefstatistic(self):
+        searchRegion.skyRegionString = lalpulsar.SkySquare2String(
+            Alpha, Delta, AlphaBand, DeltaBand,
         )
-        searchRegion.fkdotBand[3] = 0
+        searchRegion.refTime = self.tref
+        # frequency and spindowns
+        searchRegion.fkdot = np.zeros(lalpulsar.PULSAR_MAX_SPINS)
+        searchRegion.fkdotBand = np.zeros(lalpulsar.PULSAR_MAX_SPINS)
+        for k in range(3):
+            Fk = "F{:d}".format(k)
+            if Fk in range_keys:
+                searchRegion.fkdot[k] = self.search_ranges[Fk][0]
+                searchRegion.fkdotBand[k] = (
+                    self.search_ranges[Fk][1] - self.search_ranges[Fk][0]
+                    if len(self.search_ranges[Fk]) >= 2
+                    else 0.0
+                )
+        # now construct DopplerFullScan from searchRegion
         scanInit = lalpulsar.DopplerFullScanInit()
         scanInit.searchRegion = searchRegion
         scanInit.stepSizes = lalpulsar.PulsarDopplerParams()
         scanInit.stepSizes.refTime = self.tref
-        scanInit.stepSizes.Alpha = self.Alphas[-1] if len(self.Alphas) == 3 else 0.001
-        scanInit.stepSizes.Delta = self.Deltas[-1] if len(self.Deltas) == 3 else 0.001
-        scanInit.stepSizes.fkdot = np.array([0, 0, 0, 0, 0, 0, 0])
-        scanInit.stepSizes.fkdot[0] = self.F0s[-1] if len(self.F0s) == 3 else 0.0
-        scanInit.stepSizes.fkdot[1] = self.F1s[-1] if len(self.F1s) == 3 else 0.0
-        scanInit.stepSizes.fkdot[2] = self.F2s[-1] if len(self.F2s) == 3 else 0.0
+        scanInit.stepSizes.Alpha = (
+            self.search_ranges["Alpha"][-1]
+            if len(self.search_ranges["Alpha"]) == 3
+            else 0.001
+        )
+        scanInit.stepSizes.Delta = (
+            self.search_ranges["Delta"][-1]
+            if len(self.search_ranges["Delta"]) == 3
+            else 0.001
+        )
+        scanInit.stepSizes.fkdot = np.zeros(lalpulsar.PULSAR_MAX_SPINS)
+        for k in range(3):
+            if Fk in range_keys:
+                Fk = "F{:d}".format(k)
+                scanInit.stepSizes.fkdot[k] = (
+                    self.search_ranges[Fk][-1]
+                    if len(self.search_ranges[Fk]) == 3
+                    else 0.0
+                )
         scanInit.startTime = startTime
         scanInit.Tspan = float(endTime - startTime)
         scanState = lalpulsar.InitDopplerFullScan(scanInit)
+        # now obtain the PulsarSpinRange extended over all relevant Doppler shifts
         spinRangeRef = lalpulsar.PulsarSpinRange()
         lalpulsar.GetDopplerSpinRange(spinRangeRef, scanState)
+        # optional: binary parameters
+        if "asini" in range_keys:
+            if len(self.search_ranges["asini"]) >= 2:
+                maxOrbitAsini = self.search_ranges["asini"][1]
+            else:
+                maxOrbitAsini = self.search_ranges["asini"][0]
+        else:
+            maxOrbitAsini = 0.0
+        if "period" in range_keys:
```

Comment:
Also, it's taken directly from lalpulsar as is..

(For the record, I always dislike using period instead of frequency, exactly because it turns the logic upside down. ;) )